src/query/v2/plan/operator.cpp

```diff
@@ -462,6 +463,202 @@ class DistributedScanAllAndFilterCursor : public Cursor {
   std::optional<std::vector<Expression *>> filter_expressions_;
 };
 
+class DistributedScanAllByPrimaryKeyCursor : public Cursor {
+ public:
+  explicit DistributedScanAllByPrimaryKeyCursor(
+      Symbol output_symbol, UniqueCursorPtr input_cursor, const char *op_name,
+      std::optional<storage::v3::LabelId> label,
+      std::optional<std::pair<storage::v3::PropertyId, Expression *>> property_expression_pair,
+      std::optional<std::vector<Expression *>> filter_expressions, std::optional<std::vector<Expression *>> primary_key)
+      : output_symbol_(output_symbol),
+        input_cursor_(std::move(input_cursor)),
+        op_name_(op_name),
+        label_(label),
+        property_expression_pair_(property_expression_pair),
+        filter_expressions_(filter_expressions),
+        primary_key_(primary_key) {
+    ResetExecutionState();
+  }
+
+  enum class State : int8_t { INITIALIZING, COMPLETED };
+
+  using VertexAccessor = accessors::VertexAccessor;
+
+  bool MakeRequest(RequestRouterInterface &request_router, ExecutionContext &context) {
+    {
+      SCOPED_REQUEST_WAIT_PROFILE;
+      std::optional<std::string> request_label = std::nullopt;
+      if (label_.has_value()) {
+        request_label = request_router.LabelToName(*label_);
+      }
+      current_batch_ = request_router.ScanVertices(request_label);
```

Comment:
Why do we do a `ScanVertices`? Is this present just because of `PullMultiple`? If so, then please remove `PullMultiple` and all the other functions that are used only from `PullMultiple`. As this PR is not intended to implement the `MultiFrame` approach, it is better not to have any implementation. In that case the execution of `PullMultiple` will fail with an assert, which is fine.

Reply:
I will remove everything that is `Multiframe` related.